### ISSUE TITLE
ValidatorControlDelegate is only possible when using `ValidatorTextField`

### ISSUE THREAD

**Issue by markst** [ustwo/formvalidator-swift#54]
I use https://github.com/CosmicMind/Material
Which has it's own styled `UITextField` subclasses. 

My subclassed text field implements `ValidatorControl` protocol. 
```swift
class ValidationTextField: ErrorTextField, ValidatorControl
```

It would be good if I could use `ValidatorControlDelegate` when using other classes besides the provided `ValidatorTextField`.

**Comment by ghost** [comment]
The `ValidatorControlDelegate` does not require the use of `ValidatorTextField`, only a class that conforms to the `ValidatorControl` protocol. Your custom `ValidationTextField` does this. Are you having difficulty building an implementation using the `ValidatorControlDelegate` that uses your `ValidationTextField`?

Reference:

- [ValidatorControlDelegate](https://github.com/ustwo/formvalidator-swift/blob/a096e26421c938b06c6e1acb905b18b592f8e998/Sources/Protocols/ValidatorControl.swift#L12-L18)

**Comment by markst** [comment]
@aamctustwo thanks for response, I'm not sure I'm convinced.

The logic for `ValidatorControlDelegate.validatorControlDidChange` is seems to be within the `UITextField` subclass `ValidatorTextField` - which relies on subscribing to `UITextFieldTextDidChange` notification:
https://github.com/ustwo/formvalidator-swift/blob/master/Sources/Controls/ValidatorTextField-UIKit.swift#L73

**Comment by ghost** [comment]
You should be able to do something similar with your `ValidationTextField`. Since `ErrorTextField` is just a descendant subclass of `UITextField` ultimately. You may be able to tie into some of CosmicMind Material's other delegates for `TextField`, but you'd have to investigate that library further.

- ErrorTextField: https://github.com/CosmicMind/Material/blob/5e81df5a0c3448b9c811d82e57be02e57abd59f5/Sources/iOS/ErrorTextField.swift#L33
- TextField: https://github.com/CosmicMind/Material/blob/5e81df5a0c3448b9c811d82e57be02e57abd59f5/Sources/iOS/TextField.swift#L66

**Comment by markst** [comment]
Exactly. In which case how would you feel about exposing `ValidatorTextFieldResponder` to make this possible without having to duplicate a lot of logic?

@aamctustwo

**Comment by ghost** [comment]
As it is originally written, the `ValidatorTextFieldResponder` and its corresponding implementations for text views and for AppKit were tightly coupled to the core UI component (`ValidatorTextField`) and never intended to be publicly exposed.

If you'd like to raise a PR with an implementation, I'd be happy to take a look at it.
